packages/core/src/lib/upload.ts

```diff
@@ -20,19 +18,19 @@ export async function upload(
   }
 
   const { apiKey, server, organization, project } = options.upload;
-  const { outputDir } = options.persist;
-  const report = reportSchema.parse(
-    JSON.parse(readFileSync(join(outputDir, 'report.json')).toString()),
-  );
+  const report = await loadReport({
+    outputDir: options.persist.outputDir,
+    filename: options.persist.filename,
+    format: 'json',
+  });
 
+  const reportJson = JSON.parse(report);
   const data = {
     organization,
     project,
     commit: await latestHash(),
-    ...jsonToGql(report),
+    ...jsonToGql(reportJson),
   };
 
-  return uploadFn({ apiKey, server, data }).catch(e => {
-    throw new Error('upload failed. ' + e.message);
-  });
+  return uploadFn({ apiKey, server, data });
```

Comment:
If understand this change correctly, then you removed our validation of the report file (`JSON.parse` and then no `reportSchema.parse` => no safety) :warning: 

If you want to use a `loadReport` function, then I would expect that function returns an already validated `Report` instead of a plain `string`.